### PR TITLE
Enable NativeAOT in the CI

### PR DESCRIPTION
**Pull request by MichalStrehovsky**

### CODE REVIEW

**Comment by MichalStrehovsky** [comment]
@dotnet/runtime-infrastructure this enables NativeAOT testing in the CI.

This currently uses the same strategy we used in runtimelab (just run testing on the build machine). This has been fine in runtimelab because if we pass the ~20 smoke tests, the whole product is like 98% workable. We'll obviously need to up the testing if we want to ship this in some form, so I consider this strategy temporary and I'll want to use Helix. Also we have a pre-existing testing hole on ARM64. We haven't broken any of our existing customers who rely on ARM64 mostly because codegen is extensively tested in CoreCLR proper and there's very few NativeAOT-specific arch-specific issues.

The build config tries to be as lightweight as possible (not building the rest of CoreCLR, which explains the corerun-related test change in this pull request). The full build+test pass finishes in ~30 minutes on Unixes. I would like to improve that some more (e.g. we're still building superpmi, or Microsoft.Extensions.* libraries in these legs, but we don't have to).

File: src/tests/Common/tests.targets [hunk 1]
```diff
@@ -69,7 +69,10 @@
 
       <CorerunExecutable Condition="'$(RunningOnUnix)' == 'true'"  >$(CORE_ROOT)\corerun</CorerunExecutable>
       <CorerunExecutable Condition="'$(RunningOnUnix)' != 'true'">$(CORE_ROOT)\corerun.exe</CorerunExecutable>
-      <CorerunExecutable Condition="'$(TargetOS)' == 'Browser' Or '$(TargetOS)' == 'Android' or '$(TargetOS)' == 'iOSSimulator'"> $(DotnetRoot)/dotnet</CorerunExecutable>
+      <DotnetExecutable>dotnet --roll-forward LatestMajor</DotnetExecutable>
```
Comment (safern) [inline src/tests/Common/tests.targets:72]: Why is this needed?

Comment (MichalStrehovsky) [reply src/tests/Common/tests.targets:72]: Getting

```
  dotnet C:\git\runtime2\artifacts\tests\coreclr\windows.x64.Debug\Tests\Core_Root\xunit\xunit.console.dll .\nativeaot\SmokeTests\nativeaot.SmokeTests.XUnitWrapper.dll -parallel collections -html C:\git\runtime2\artifacts\log\TestRun.html -xml C:\git\runtime2\artifacts\log\TestRun.xml  -notrait category=outerloop -notrait category=failing -nocolor
  Microsoft.DotNet.XUnitConsoleRunner v2.5.0 (64-bit .NET Core 3.1.21)
     nativeaot.SmokeTests.XUnitWrapper: Exception thrown during test framework discoverer construction: System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
```

locally without that. Not sure why it's choosing a 3.1 version of the runtime to run xunit. The host resolution is a mystery.

Comment (jkoritzinsky) [reply src/tests/Common/tests.targets:72]: It looks like it's trying to pick the host to match `xunit.console.dll` since there's no `-runtimeconfig` parameter to point to a different `runtimeconfig.json`. Since `xunit.console.dll` is probably built against .NET Core 3.1, that's what it selects for the runtime without `LatestMajor` roll-forward.

Comment (MichalStrehovsky) [reply src/tests/Common/tests.targets:72]: Thank you @jkoritzinsky! Any preference for solving this or is roll forward okay?

Alternatively, I could change these legs to build full CoreCLR and then corerun is available and this change is not needed. But felt odd having to build entire CoreCLR just to run xunit. I guess we could also AOT compile the xunit runner, but that sounds like a lot of additional build complexity.

Comment (jkoritzinsky) [reply src/tests/Common/tests.targets:72]: I’m fine keeping the roll-forward for now as we can/should probably use the new xunit wrapper generator instead of the console runner if we want to AOT this.

Can you add a comment here though about the reason for the roll-forward so we don’t forget?

**Review by jkoritzinsky — APPROVED** [review]
